src/main/java/org/truffleruby/language/arguments/RubyArguments.java

```diff
@@ -168,25 +183,68 @@ public static Object getBlock(Frame frame) {
         return block;
     }
 
-    public static int getArgumentsCount(Frame frame) {
-        return frame.getArguments().length - RUNTIME_ARGUMENT_COUNT;
+    public static Object getKeywordArgumentsDescriptor(Frame frame) {
+        final Object keywordArgumentsDescriptor = frame.getArguments()[ArgumentIndicies.KEYWORD_ARGUMENTS_DESCRIPTOR
+                .ordinal()];
+        assert keywordArgumentsDescriptor != null;
+        return keywordArgumentsDescriptor;
+    }
+
+    public static KeywordDescriptor getKeywordArgumentsDescriptorUnsafe(Frame frame) {
+        final KeywordDescriptor keywordDescriptor = (KeywordDescriptor) frame
+                .getArguments()[ArgumentIndicies.KEYWORD_ARGUMENTS_DESCRIPTOR.ordinal()];
+        assert keywordDescriptor != null;
+        return keywordDescriptor;
+    }
+
+    public static Object getDescriptorKeywordLength(KeywordDescriptor descriptor) {
```

Comment:
Let's inline this. Also it should return `int` so no need for casting.

Reply:
@nirvdrum

Reply:
Resolved in 46539aebf0.